src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -339,11 +340,17 @@ public List<TOutput> ConvertAll<TOutput>(Converter<T, TOutput> converter)
             }
 
             List<TOutput> list = new List<TOutput>(_size);
-            for (int i = 0; i < _size; i++)
+            list._size = _size;
+            // Use Span to avoid covarience checks from using List
+            Span<TOutput> output = CollectionsMarshal.AsSpan(list);
```

Comment:
I don't see why `List`'s own methods need to use `CollectionMarshal`. `new Span(_items, _size)` should be enough.

Reply:
🤔  That's a good point 😄